code/game/objects/items/pet_carrier.dm

```diff
@@ -146,6 +146,8 @@
 	cut_overlay("locked")
 	if(open)
 		icon_state = initial(icon_state)
+	else if(/obj/item/pet_carrier/xenobio)
```

Comment:
Override this proc in the subtype,don't do stuff like this. Its a violation of OOP

Reply:
fixed